packages/xo-server-sdn-controller/src/utils/tls-helper.js

```diff
@@ -0,0 +1,55 @@
+import fromEvent from 'promise-toolbox/fromEvent'
+import createLogger from '@xen-orchestra/log'
+import { connect } from 'tls'
+
+// =============================================================================
+
+const log = createLogger('xo:xo-server:sdn-controller:tls-connect')
+
+// =============================================================================
+
+export class TlsHelper {
+  updateCertificates(clientKey, clientCert, caCert) {
+    this._clientKey = clientKey
+    this._clientCert = clientCert
+    this._caCert = caCert
+    log.debug('Certificates have been updated')
+  }
+
+  // ---------------------------------------------------------------------------
+
+  async connect(address, port) {
+    const options = {
+      ca: this._caCert,
+      key: this._clientKey,
+      cert: this._clientCert,
+      host: address,
+      port,
+      rejectUnauthorized: false,
+      requestCert: false,
+    }
+    const socket = connect(options)
+    try {
+      await fromEvent(socket, 'secureConnect', {})
+    } catch (error) {
+      log.error('TLS connection failed', {
+        error,
+        code: error.code,
```

Comment:
What's the point?

Reply:
The point of what?

Reply:
Of logging `error.code` in addition to `error`.

Reply:
Useless indeed... good point! ^^ I'll remove it.